soh/soh/Enhancements/randomizer/3drando/fill.cpp

```diff
@@ -485,20 +485,18 @@ static void GeneratePlaythrough() {
     GetAccessibleLocations(ctx->allLocations, SearchMode::GeneratePlaythrough);
 }
 
-RandomizerArea LookForExternalArea(Area* curRegion, std::vector<RandomizerRegion> alreadyChecked){//RANDOTODO curREGION
-  for (auto& entrance : curRegion->entrances) {
+RandomizerArea LookForExternalArea(const Area* const currentRegion, std::vector<RandomizerRegion> &alreadyChecked){
+  for (const auto& entrance : currentRegion->entrances) {
     RandomizerArea otherArea = entrance->GetParentRegion()->GetArea();
-    if (otherArea != RA_LINKS_POCKET){ //if it's links pocket, do not propogate this, link's pocket is not a real Area
-      if(otherArea != RA_NONE){
-        return otherArea;
-        //if the area hasn't already been checked, check it
-      } else if (std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()) {
+    if(otherArea == RA_NONE && std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()){
```

Comment:
Hey! First of all, thanks for applying given recommendations!
This line could be simplified by moving std::find check to separate boolean evaluation expression.
Like this:
```
const bool isAreaAlreadyChecked = std::ranges::find(alreadyChecked,
                                                    entrance->GetParentRegionKey()) == alreadyChecked.end();
if (otherArea == RA_NONE && !isAreaAlreadyChecked) {
```

Reply:
I have no more comments beside this one, fix that, and it's good to go!